### PR TITLE
Add spellchecker and fix spelling

### PR DESCRIPTION
**Pull request by Grayray75**
Uses [CSpell](https://cspell.org/) as an GitHub action to check for spelling mistakes.
Also fixes a lot of spelling mistakes in yarn.

### CODE REVIEW

**Review by BleachDev — APPROVED** [review]
there do be a lot of typos

File: cspell.config.json [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        "regen",
+        "reloadable",
+        "renderable",
+        "renderbuffer",
+        "resourcepack",
+        "respawn",
+        "rgba",
+        "rtmp",
+        "schoolable",
+        "ScreenHandlerProprtyUpdateS2CPacket",
```
Comment (thecatcore) [inline cspell.config.json:144]: Proprty?

**Review by thecatcore — APPROVED** [review]

File: cspell.config.json (hunk 1 above)
Comment (Grayray75) [reply cspell.config.json:144]: Problem is that currently there are two classes with the same name: `ScreenHandlerProprtyUpdateS2CPacket` and  `ScreenHandlerPropertyUpdateS2CPacket`.
That's why I couldn't fix the spelling and added it to the ignore list.

Comment (thecatcore) [reply cspell.config.json:144]: The new system allows having multiple class mappings in the same file so you can probably move the misspelled one in the correct file and rename the mapping at the same time.

Comment (Grayray75) [reply cspell.config.json]: I didn't know that this was possible

**Pull request merged**